js/app/src/Blocks.svelte

```diff
@@ -240,12 +240,30 @@
 		id: ++_error_id
 	});
 	let _error_id = -1;
+
+	let user_left_page: boolean = false;
+	document.addEventListener("visibilitychange", function () {
+		if (document.visibilityState === "hidden") {
+			user_left_page = true;
+		}
+	});
+
 	const MESSAGE_QUOTE_RE = /^'([^]+)'$/;
 
 	const DUPLICATE_MESSAGE =
 		"There is a long queue of requests pending. Duplicate this Space to skip.";
+	const MOBILE_QUEUE_WARNING =
+		"On mobile, the connection can break if this tab is unfocused or the device sleeps, losing your position in queue.";
+	const MOBILE_RECONNECT_MESSAGE =
+		"Lost connection due to leaving page. Rejoining queue...";
 	const SHOW_DUPLICATE_MESSAGE_ON_ETA = 15;
+	const SHOW_MOBILE_QUEUE_WARNING_ON_ETA = 10;
```

Comment:
Perhaps also set this to 15 seconds so that we have a consistent definition of a "long waiting time" that we can communicate to users

Reply:
Makes sense

Reply:
I chose 10 seconds specifically because that's how long it seems iPhones break a websocket connection on low power mode. I don't think there's any need for these to match

Reply:
That should probably be documented in a comment near the constant :)